### PR TITLE
Update to CICE-Consortium main #422117fc5d Oct 9, 2022

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update to CICE-Consortium main #422117fc5d Oct 9, 2022
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    This is bit-for-bit with #422117fc5d on consortium main, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#14c8157acaca854d438a2f6dee58ca5edea48317
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit with #422117fc5d
    - [ ] different at roundoff level
    - [X] more substantial, not bit-for-bit with latest E3SMIcepackDEV
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [ ] No
- Does this PR add any new test cases?
    - [X] Yes
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

This bring in
- Update dEdd implementation in Icepack (changes answers, to be consistent with E3SM branch)
- Refactored evp subcycling loop (bit-for-bit)
- Move deformation computation out of B subcycling loop (changes answers) 
- Update grid averaging for tmass, aice, uvelT, vvelT (changes answers)
- Add variable grid spacing box configuration
- Update Pull Request Template and documentation

### CODE REVIEW

**Comment by apcraig** [comment]
I have confirmed this merge now makes the E3SMIcepackDEV branch bit-for-bit with the current Consortium main, #422117fc5d.  This also brings E3SMIcepackDEV fully up to date with CICE Consortium main #422117fc5d.

File: doc/source/user_guide/ug_case_settings.rst [hunk 1]
```diff
@@ -249,7 +249,9 @@ grid_nml
    "", "``pop``", "pop thickness file in cm in ascii format", ""
    "``close_boundaries``", "logical", "force two gridcell wide land mask on boundaries for rectangular grids", "``.false.``"
    "``dxrect``", "real", "x-direction grid spacing for rectangular grid in cm", "0.0"
+   "``dxscale``", "real", "user defined rectgrid x-grid scale factor", "1.0"
    "``dyrect``", "real", "y-direction grid spacing for rectangular grid in cm", "0.0"
+   "``dyscale``", "real", "user defined rectgrid y-grid scale factor", "1.0"
```
Comment (eclare108213) [inline doc/source/user_guide/ug_case_settings.rst:254]: Is the variable rectgrid scaling not described elsewhere in the docs?  I don't see it.  Would be good to add a brief description somewhere, maybe in one of the sections that mention the box tests like 3.3.1.3.1. box2001.

**Review by eclare108213 — APPROVED** [review]
My comment needs to be done in CICE main before E3SMIcepackDEV - just making a note for followup.  It only needs a sentence...

File: doc/source/user_guide/ug_case_settings.rst (hunk 1 above)
Comment (apcraig) [reply doc/source/user_guide/ug_case_settings.rst:254]: Agree this should be addressed on Consortium main and should have been addressed better when the variable grid stuff was added.  I have added an issue so we don't forget, https://github.com/CICE-Consortium/CICE/issues/770.

**Pull request merged**